lib/active_merchant/billing/gateways/komoju.rb

```diff
@@ -0,0 +1,84 @@
+require 'json'
+require 'komoju'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class KomojuGateway < Gateway
+      self.live_url = "https://gateway-sandbox.degica.com/api/v1"
+      self.supported_countries = ['JP']
+      self.default_currency = 'JPY'
+      self.money_format = :cents
+      self.homepage_url = 'https://komoju.com/'
+      self.display_name = 'Komoju'
+      self.supported_cardtypes = [:visa, :master, :american_express, :jcb]
+
+      def initialize(options = {})
+        requires!(options, :login)
+        @api_key = options[:login]
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        params = {
+          amount: amount(money),
+          description: options[:description],
+          payment_details: payment_details(payment, options),
+          currency: options[:currency] || currency(money),
+          external_order_num: options[:order_id], # should be removed when the param becomes optional
+          metadata: {
+            order_id: options[:order_id]
+          }
+        }
+        commit(params)
+      end
+
+      private
+
+      def payment_details(payment, options)
+        details = if payment.is_a?(CreditCard)
```

Comment:
After discussing this we decided that for now, a hash is probably the simplest way to do this to avoid integration issues (hash is easier than requiring a special model to be passed in).